app/lib/frontend/handlers/account.dart

```diff
@@ -2,13 +2,31 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'package:client_data/account_info.dart';
 import 'package:shelf/shelf.dart' as shelf;
 
 import '../../account/backend.dart';
+import '../../frontend/backend.dart';
 import '../../shared/handlers.dart';
 
 /// Handles /api/account/v1/info
 Future<shelf.Response> accountInfoHandler(shelf.Request request) async {
```

Comment:
This is not a good REST API...

We don't need to request the email from `pub.dev` we already know that client-side from the login session.

I thought we only had this to prove that we could do authentication... 
but I guess this partially serves to ensure the `User` entity is created on the server?

-----

hmm, IMO the best way to replace this would be to build an end-point that returns the list of packages for which you are an uploader.

I see how that will make publishers harder, and it could grow large for some users...

Reply:
I agree about the e-mail part, kept it because it would allow us a bit of debugging in case we needed it, but of course the rest of the payload is unrelated.

I think returning all the packages is problematic on many parts, e.g. it is a larger operation on the Datastore as well, and most of its data gets discarded anyway. What we need is a user-scope view of a package (e.g. it contains its uploader status, later it could contain whether the package is starred or not). Some options:
- We could extend some of the current API endpoints that return e.g. the current version request could include the uploader status, e.g. https://pub.dev/api/packages/pana/versions/0.12.18
- We could augment the `/api/packages/pana/*` path with `/user` to fetch data for the current user (similar to the `/metrics` which returns the current analysis summary).
- We could do a new path `/api/account/packages/<package>`?

Reply:
>  and most of its data gets discarded anyway.

Not really, imagine we'll what to show this in the menu when signed-in... maybe not just now, but someday.

It would also be a generically useful API end-point, for the CLI or other integrators.

-----
But good point about wanting to expose whether it's starred! :)

I do like the idea that we put it under `/api/account/...` then... as it's user-specific information (requires you to be signed-in).

This could also have email notification preferences, should we ever add this. 

So this is perhaps user-package-options or user-package-preferences... Just brainstorming here..

 * `GET|PUT /api/account/preferences/packages/<package>`, or,
 * `GET|PUT /api/packages/<package>/user-options`

Reply:
These seems to be good to me:
- `/api/account/preferences` - global user preferences (e.g. show/hide e-mail)
- `/api/account/preferences/packages/<package>` - package-specific info (e.g. isUploader) and preferences (e.g. stars)

I have a slight dislike for communicating a read-only status of `isUploader` through "preferences", but I think it is be a good-enough term for it. `fields` or `attributes` doesn't sound any better, does it?

Reply:
I always imagined stars would be something like:

 * `GET|PUT /api/user/<userId>/stars/<package>`

but I'm starting to think that `<userId>` in URLs is a bad idea :)

Maybe we call it "options"  instead?

`GET|PUT /api/account/options/package/<package>`

Reply:
I've settled with `/api/account/options/packages/<package>`, it is pointing towards more consistency with the `/api/packages/<package>/options` (also note: the plural in packages).